TombEngine/Scripting/Internal/TEN/Misc/Miscellaneous.cpp

```diff
@@ -393,6 +396,31 @@ namespace Misc
 		TENLog(message, level, LogConfig::All, USE_IF_HAVE(bool, allowSpam, false));
 	}
 
+	/// Get the projected 2D screen position of a 3D world position. Returns an invalid distant 2D position if the 3D position is behind the camera.
+	//@tparam Vec3 pos3D 3D world position.
+	//@return Vec2 pos2D Projected 2D screen position. NOTE: Screen space resolution is 100x100.
+	//@usage 
+	//
+	// Example: display string at an object's position.
+	// local string = DisplayString('Example', 0, 0, Color(255, 255, 255), false)
+	// local pos2D = Get2DPosition(Lara:GetPosition())
+	// string:SetPosition(PercentToScreen(pos2D.x, pos2D.y))
+	static Vec2 Get2DPosition(const Vec3& pos)
+	{
+		// TODO: Move to Renderer11Enums.h.
+		constexpr auto INVALID_2D_POSITION = Vector2(FLT_MAX);
```

Comment:
I think that instead of returning nonsensical 2D position, you should also return `nil` to Lua in such case. That is, make `Get2DPosition` lua method to return `optional` too.